src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -331,7 +331,28 @@ public function symlink($originDir, $targetDir, $copyOnWindows = false)
             }
         }
 
-        if (!$ok && true !== @symlink($originDir, $targetDir)) {
+        if (!$ok) {
+            $ok = @symlink($originDir, $targetDir);
+        }
+
+        if (!$ok && strncasecmp(PHP_OS, 'WIN', 3) == 0) {
```

Comment:
Minor comment: to detect if the OS is macOS, we use the `PHP_OS` constant as follows: `'Darwin' === PHP_OS` But for Windows, we always use this indirect detection: `'\\' === DIRECTORY_SEPARATOR`